src/repository/resources/chain.rs

```diff
@@ -610,11 +610,11 @@ impl<T: Block> iter::FromIterator<T> for OwnedChain<T> {
                     *res.last_mut().unwrap() = T::new(last_min, block.max())
                 }
                 else {
-                    res.push(block)
+                    res.push(T::new(block.min(), block.max()))
```

Comment:
So, this will ensure that when this is used for T = IpBlock the block will be recreated and check whether it should be a Prefix or a Range. This will fix the encoding issue we found when adding a range that should have been a prefix.

Perhaps add a comment as this is not immediately apparent?

Reply:
Fair point. I’ll add some explanation.